### PR TITLE
Minor changes to magnetic field in preparation for AlongStep integration

### PR DESCRIPTION
**Pull request by sethrj**
The main refactoring here is the ability to create a field propagator much more simply:
```c++
    UniformField field({0, 0, test.field_value});
    auto propagate = make_mag_field_propagator<DormandPrinceStepper>(
        field, driver_options, particle_track, &geo_track);
```

- da6c4613 Mostly replace magfieldtraits with helper class
- 1acd0a38 More documentation tweaks
- 21097a9b Rename MagField to Field
- 7f7b5310 Forward declare and IWYU
- 4848d82b Add perfect forwarding and const correctness
- e2b7f897 Minor code and documentation improvements
- 5283bc66 Rename function-like objects as action verbs
- 2881814f Rename FieldParamsData to FieldDriverOptions

### CODE REVIEW

**Comment by whokion** [comment]
@sethrj there are several compilation errors from cu files.

**Review by whokion — CHANGES_REQUESTED** [review]
Please fix remaining compilation errors from UserParamField.test.cu
and then it should be good to go.

File: test/celeritas/field/FieldDriver.test.cu [hunk 1]
```diff
@@ -34,24 +32,22 @@ namespace celeritas_test
...
 
-    // Construct the driver
-    UniformMagField field({0, 0, test_params.field_value});
-    using RKTraits = MagTestTraits<UniformMagField, DormandPrinceStepper>;
-    RKTraits::Equation_t equation(field, units::ElementaryCharge{-1});
-    RKTraits::Stepper_t  rk4(equation);
-    RKTraits::Driver_t   driver(data, &rk4);
+    auto driver = make_mag_field_driver<DormandPrinceStepper>(
+        UniformField({0, 0, test_params.field_value}),
+        field_params,
```
[the remark this answers is not in the document]
Comment (sethrj) [reply test/celeritas/field/FieldDriver.test.cu:49]: Sorry copy-paste error and I didn't test the cuda code .. should have left this as draft

**Comment by sethrj** [comment]
@whokion I fixed the bug—unintentional dangling reference. Thanks for the review!

**Pull request merged**